pkg/aws/aws_client/role.go

```diff
@@ -292,3 +305,52 @@ func completeRolePolicyDocument(statement map[string]interface{}) (string, error
 	assumeRolePolicyDocument, err := json.Marshal(rolePolicyDocument)
 	return string(assumeRolePolicyDocument), err
 }
+
+func (client *AWSClient) AttachPolicy(roleName string, policyArn string) error {
+	policyAttach := iam.AttachRolePolicyInput{
+		PolicyArn: &policyArn,
+		RoleName:  &roleName,
+	}
+	_, err := client.IamClient.AttachRolePolicy(context.TODO(), &policyAttach)
+	if err != nil {
+		return err
+	}
+	timeout := 2
+	start := 0
+
+	for start < timeout {
+
+		if attached, _ := client.PolicyAttachedToRole(roleName, policyArn); attached {
+			return nil
+		}
+		time.Sleep(1 * time.Minute)
+		start++
+	}
+	return err
+}
+
+func (client *AWSClient) PolicyAttachedToRole(roleName string, policyArn string) (bool, error) {
```

Comment:
@OAharoni-RedHat @xueli181114 @yingzhanredhat Where are we on unit testing for this client? We really need to start adding it soon as this client is continuing to grow quickly without any unit test coverage.

Reply:
Definitely agree that there should be some unit testing added to this client, however I personally think it's maybe a bit out of scope for this current lift and shift task. Would be happy to create some new Jira cards for working on unit tests for the client to add to the backlog, does that seem like a reasonable compromise to you? Once the frameworks and everything have been set up we can add some blockers on PRs to ensure x% of the new code has unit tested to ensure we stay on top of things